core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushJoinIntoTableScan.java

```diff
@@ -99,6 +100,19 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
             return Result.empty();
         }
 
+        if (getJoinPushdownMode(context.getSession()) == JoinPushdownMode.AUTOMATIC) {
+            PlanNodeStatsEstimate joinStats = context.getStatsProvider().getStats(joinNode);
+            PlanNodeStatsEstimate leftStats = context.getStatsProvider().getStats(joinNode.getLeft());
+            PlanNodeStatsEstimate rightStats = context.getStatsProvider().getStats(joinNode.getRight());
+            if (joinStats.isOutputRowCountUnknown() || leftStats.isOutputRowCountUnknown() || rightStats.isOutputRowCountUnknown()) {
```

Comment:
What if one of left output count or right count is know and larger than join output row count, why not pushdown join in such case as well ?

Reply:
Yeah - we could. Though it is strictly theoretical case. As if we do not know either left or right size. We would not know the join size :)

Reply:
Ah right, I missed that. How about doing something similar to `DetermineJoinDistributionType#getSourceTablesSizeInBytes` just in case the source table has stats but after some node in between the output stats become unknown. Also, any particular reason for basing this on row count instead of size ?

Reply:
Not really. Probably size would be more appropriate. I will see how painful it is to change that.